### PR TITLE
Fallback to old image search if new one fails

### LINKED ISSUE #12: bug: if you search "pull up fat stogie with the boys" the image doesn't work
image

### PR DESCRIPTION
**Pull request by nickells**
Fixes #12

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +21,7 @@
     "stopword": "^0.2.4"
   },
   "devDependencies": {
-    "eslint": "^6.0.1"
+    "eslint": "^6.0.1",
+    "nodemon": "^2.0.6"
```
Comment (nickells) [inline package.json:25]: allows me to watch files and restart the server when one changes

File: server/sendScreenshot.js [hunk 2]
```diff
@@ -22,6 +22,24 @@ const getBufferFromRequest = (url) => {
   })
 }
 
+const getCroppedScreenshot = async (page, firstImageUrl) => {
```
Comment (nickells) [inline server/sendScreenshot.js:25]: mostly copy-pasted from https://github.com/nickells/jeremy-the-bot/commit/417be78800c7498bb0107e49e6722300810cc92b#diff-01a6cefc3183806d1dee00bf0c4ec1bed99a97bb1383f8519f433612fcdf4d72

File: server/sendScreenshot.js [hunk 3]
```diff
@@ -41,9 +59,14 @@ const sendScreenshot = async (event, query, firstImageOnly) => {
...
-  } else {
-    data = await page.screenshot()
+    try {
+      data = await getBufferFromRequest(firstImageUrl)
+    }
+    catch (e) {
+      console.warn('error fetching', e)
+      data = await getCroppedScreenshot(page, firstImageUrl)
+    }
+    if (data.length === 0) data = await getCroppedScreenshot(page, firstImageUrl)
```
Comment (nickells) [inline server/sendScreenshot.js:69]: if the first one errors or returns an empty buffer, do the fallback

File: package-lock.json [hunk 4]
```diff
@@ -24,6 +24,12 @@
         "js-tokens": "^4.0.0"
```
Comment (nickells) [inline package-lock.json:24]: automatically generated, ignore this

**Comment by jonkislin** [comment]
codelord @nickells says it works so I believe him teehee

**Pull request merged**